src/test/scala/mesosphere/marathon/state/AppDefinitionTest.scala

```diff
@@ -119,10 +128,12 @@ class AppDefinitionTest extends MarathonSpec with Matchers {
     )
     val result1 = AppDefinition().mergeFromProto(app1.toProto)
     assert(result1 == app1)
+    validateJsonSchema(app1)
 
     val app2 = AppDefinition(cmd = None, args = Some(Seq("a", "b", "c")))
     val result2 = AppDefinition().mergeFromProto(app2.toProto)
     assert(result2 == app2)
+    validateJsonSchema(app2, false) // no id
   }
 
   test("Validation") {
```

Comment:
this is a very long test.

It would be great to break that apart so one could see all failing conditions at once when something breaks.

You can even write your own helper method that generates the tests since "test" is just a method call.